app/test/features/calendar_sync/calendar_sync_test.dart

```diff
@@ -115,4 +124,108 @@ void main() {
       expect(targetEvent.reminders?.first.minutes, 10);
     });
   });
+
+  group('Calendar Sync Provider test', () {
+    testWidgets('by default sync all', (tester) async {
+      SharedPreferences.setMockInitialValues({}); // no values set yet.
+
+      final roomAEvents = generateMockCalendarEvents(count: 5, roomId: 'roomA');
+      final roomBEvents = generateMockCalendarEvents(count: 5, roomId: 'roomB');
+      final roomCEvents = generateMockCalendarEvents(count: 5, roomId: 'roomC');
+      final List<MockCalendarEvent> events = [];
+      final client = MockClient();
+      final mockRoom = MockRoom();
+      final Stream<bool> updateSteam = Stream.empty();
+      when(() => client.subscribeModelObjectsStream(any(), any()))
+          .thenAnswer((a) => updateSteam);
+      when(() => mockRoom.userSettings())
+          .thenAnswer((_) async => MockRoomUserSettings());
+      events.addAll(roomAEvents);
+      events.addAll(roomBEvents);
+      events.addAll(roomCEvents);
+
+      final container = ProviderContainer(
+        overrides: [
+          utcNowProvider.overrideWith((ref) => MockUtcNowNotifier()),
+          allEventListProvider.overrideWith((r, a) => events),
+          alwaysClientProvider.overrideWith((ref) => client),
+          maybeRoomProvider.overrideWith(
+            () => MockAlwaysTheSameRoomNotifier(
+              room: mockRoom,
+            ),
+          ),
+          calendarEventProvider.overrideWith(
+            () => MockFindAsyncCalendarEventNotifier(events: events),
+          ),
+        ],
+      );
+      addTearDown(container.dispose);
+
+      // keeping the provider alive for the tests
+      // ignore: unused_local_variable
+      final subscription = container.listen(eventsToSyncProvider, (a, b) {});
+
+      final state = await container.read(eventsToSyncProvider.future);
+      expect(
+        state.length,
+        events.length,
+      );
+    });
+    testWidgets('excluded rooms are excluded', (tester) async {
+      SharedPreferences.setMockInitialValues({}); // no values set yet.
+
+      final roomAEvents = generateMockCalendarEvents(count: 5, roomId: 'roomA');
+      final roomBEvents = generateMockCalendarEvents(count: 5, roomId: 'roomB');
+      final roomCEvents = generateMockCalendarEvents(count: 5, roomId: 'roomC');
+      final List<MockCalendarEvent> events = [];
+      final roomA = MockRoom();
+      final roomB = MockRoom();
+      final roomC = MockRoom();
+      when(() => roomA.userSettings())
+          .thenAnswer((_) async => MockRoomUserSettings());
+      when(() => roomB.userSettings()).thenAnswer(
+        // b will be ignored
+        (_) async => MockRoomUserSettings(include_cal_sync: false),
```

Comment:
as I wanted to make sure the entire flow of providers works well, I explicitly am not mocking the highest provider but the one on the lowest level: have a room with new settings mocked that do or do not get them to be included.